src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2019-12-30 20:52
+from __future__ import unicode_literals
+
+from datetime import timedelta, datetime
+
+from django.db import migrations
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Inserts Postgres events into the eventstream if there are recent events in Postgres.
+
+    This is for open source users migrating from 9.x who want to keep their events.
+    If there are no recent events in Postgres, skip the backfill.
+    """
+    from sentry import eventstore, eventstream
+    from sentry.utils.query import RangeQuerySetWrapper
+
+    Event = apps.get_model('sentry', 'Event')
+    Group = apps.get_model('sentry', 'Group')
+    Project = apps.get_model('sentry', 'Project')
+
+    days_to_check = 14
```

Comment:
So, the reason I picked a shorter range for this is that for a long time we were dual writing events to both Postgres and Snuba in earlier versions of 10.x. The code that does this was only removed in late November with https://github.com/getsentry/sentry/pull/15662. I wanted to make sure we skip the migration events for users upgrading from earlier versions of 10.x even if still they have some older events in Postgres. It's a bit of a hacky way of guessing what version they are coming from, I can't think of a better way to determine this. Any ideas?

Reply:
There were no earlier 10.x versions released except for the beta phase and even then we did not do any dual writing for on-premise as it is not very practical :)

I've tried running the backfill script over and over for the same events and it seems like idempotent so I'd say respecting the retention days value makes sense here. If it is set to `0`, it means _all events_ though so we probably would want to limit that to `90` or something.

On top of this, we can introduce a kill switch based on environment variables, which we can set when we are sure we don't need a migration. What do you think?

Reply:
Ok. So, we are dropping support for infinite retention with v10 then, and defaulting to 90. But if the user has some value set that is larger than 90 days we will still respect that?

Kill switch sounds like a good option for the v10 beta users.

Reply:
Actually what do you mean by 

> we did not do any dual writing for on-premise?

How did you avoid that?

Reply:
>But if the user has some value set that is larger than 90 days we will still respect that?

I'd cap it to 90 or at least not respect infinity, which is the default before 10 :(

>How did you avoid that?

Since the latest released version is 9.1.2, which doesn't require Snuba, we never had to do any dual writes or anything. v10 will be the first version featuring (and requiring) Snuba. This is why the migration is critical as dual-writing in on-premise is never practical: we cannot really ask people to change their config and update after X days and drop a dependency easily. From now on, we may be able to but still not an easy problem to solve when we're not controlling the deploy/production environment.